### ISSUE TITLE
Azure storage queue: Keda scales down only on visable messages

### ISSUE THREAD

**Issue by wagenrace** [kedacore/keda#1615]
**Report**

Keda scales down the pods even when invisible message are still being processed

**Expected Behavior**

Send a message.
Keda scales to 1 pods
Pod processes message.
Pod finishes
Cool down time
Keda scales down to 0 pods

**Actual Behavior**

Send a message.
Keda scales to 1 pods
Keda things queue length is 0
Cool down time
Keda scales down to 0 pods

**Steps to Reproduce the Problem**

1. Azure function app on storage queue with sleep of 10 minutes
2. Keda on message same storage queue with messageCount: "1" and cooldowntime of 300
3. Send a message, pod will never finish

**Logs from KEDA operator**

```
example
```

**KEDA Version**

2.0.0

**Kubernetes Version**

1.18

**Platform**

Microsoft Azure

**Scaler Details**

_No response_

I know I can increase the cooldown time but that still means the scaling is not reacting to new messages coming in

**Comment by tomkerkhove** [comment]
We do not guarantee that pods will run to completion given we just instruct Kubernetes how many instances we need. It's up to Kubernetes then to manage the pods, which could involve killing instances.

If your process needs to run to completion, you should consider using ScaledJob instead.

**Comment by wagenrace** [comment]
Thank you for you response, ScaledJob is indeed a better fit for our problem.
But I still see the behavior that is only scales to visible items in the storage queue.

example:
I have 1 item in the queue. It scales to 1 pod.
Well the pod is running I send a second message (so now I have 1 visible and 1 invisible item in the queue) and it does nothing because 1 item and 1 visible item is correct according to KEDA.

Once I send a third message (so now I have 2 visible and 1 invisible items in the queue) it will scale to 2 pods

It there a way to scale to all items in the queue

**Comment by tomkerkhove** [comment]
If you send the second message and wait for the polling period to pass; does it still not scale?

**Comment by wagenrace** [comment]
Nope, I changed the polling period to 5 seconds and nothing happens till the third message is introduced.

I also tried service bus (they lock message instate of making them invisible) and there it is scaling to the full queue.

Service bus is not an option for us unfortunately

**Comment by wagenrace** [comment]
I did manager to work around it by changing the scalingStatergy from default to this:

`  
scalingStrategy:
    strategy: "accurate"
    customScalingQueueLengthDeduction: 0  
    customScalingRunningJobPercentage: "0"  
`

**Comment by tomkerkhove** [comment]
Adding @TsuyoshiUshio
